src/hotspot/share/runtime/threads.cpp

```diff
@@ -833,7 +833,7 @@ jint Threads::create_vm(JavaVMInitArgs* args, bool* canTryAgain) {
 
 // Threads::destroy_vm() is normally called from jni_DestroyJavaVM() when
 // the program falls off the end of main(). Another VM exit path is through
-// vm_exit() when the program calls System.exit() to return a value or when
+// vm_exit() when
```

Comment:
I think you have to read it like this:

```
Exit Path 1: reaching from Threads::destroy_vm()

Exit Path 2: reaching from
    - vm_exit()
    - System.exit()
    - Serious error
```

Maybe add commas to disambiguate.
(https://en.wikipedia.org/wiki/Serial_comma)

Reply:
I adjusted the comment and brought back the deleted part.